src/browser/api/window.js

```diff
@@ -1590,7 +1590,7 @@ Window.show = function(identity, force = false) {
     let payload = {};
     let defaultAction = () => {
         const dontShow = (
-            browserWindow.isMinimized() ||
+            browserWindow.isMinimized() && browserWindow.isVisible() ||
```

Comment:
This can be shortened (and easier to read because of parenthesis) to
```
browserWindow.isVisible()
&&
(browserWindow.isMinimized() || browserWindow.isMaximized())
```